src/math/ITiglObjectiveFunction.cpp

```diff
@@ -66,13 +67,11 @@ void ITiglObjectiveFunction::getNumericalGradient(const double * x, double * dx)
     for (int i = 0; i < getParameterCount(); ++i) {
         xnew[i] = x[i] + h;
 
-        double ofi = getFunctionValue(xnew);
+        double ofi = getFunctionValue(&xnew.front());
```

Comment:
Is this really safe? I.e. is it defined in the standard that ```&xnew.front()``` is equivalent to an array? I see no real benefit in using a c++ array here. Okay, exception safety is a reason.
Also, this is performance critical code - we must make sure that a c++ style vector is not slower here than a simple array.